modules/internal/ChapelDomain.chpl

```diff
@@ -2988,4 +2989,116 @@ module ChapelDomain {
 
   }  // record _domain
 
+  //
+  // Support for creating domains using tuples of bounds
+  //
+
+  /* Creates a rectangular domain with bounds defined by the scalar values `low`
+      and `high`. If `inclusive` is true, the domain includes the `high` value.
+      Otherwise, the domain excludes the `high` value.
+   */
+  proc makeRectangularDomain(low: ?t1, high: ?t2, param inclusive: bool = true)
```

Comment:
You probably want to mark these overloads unstable, and then to add a corresponding test about unstable warnings to `test/unstable``.